lib/min_heap.rb

```diff
@@ -44,28 +52,65 @@ def to_s
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(1)
+  # Space complexity: O(1)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    return @store.empty?
   end
 
   private
 
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+
+
+  # Time complexity: O(logn)
+  # Space complexity: O(1)
   def heap_up(index)
```

Comment:
Since this method is recursive, you have O(log n) space complexity.  You have to account for the system stack.